l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -52,7 +52,21 @@ contract ExecutorFacet is Base, IExecutor {
             blobCommitments = _verifyBlobInformation(_newBatch.pubdataCommitments[1:], blobHashes);
         } else if (pubdataSource == uint8(PubdataSource.Calldata)) {
             // In this scenario pubdataCommitments is actual pubdata consisting of l2 to l1 logs, l2 to l1 message, compressed smart contract bytecode, and compressed state diffs
-            require(logOutput.pubdataHash == keccak256(_newBatch.pubdataCommitments[1:]), "wp");
+            require(
+                logOutput.pubdataHash ==
+                    keccak256(_newBatch.pubdataCommitments[1:_newBatch.pubdataCommitments.length - 64]),
+                "wp"
+            );
+            blobHashes[0] = bytes32(
+                _newBatch.pubdataCommitments[_newBatch.pubdataCommitments.length - 64:_newBatch
+                    .pubdataCommitments
+                    .length - 32]
+            );
```

Comment:
Can we take blobHash from `logOutput.blob1Hash` instead?

Reply:
+1 - this would save us 32 bytes right?

Reply:
good catch, just updated